editor/src/components/canvas/canvas-strategies/strategies/ancestor-metastrategy.tsx

```diff
@@ -0,0 +1,119 @@
+    if (
+      EP.isEmptyPath(targets[0]) ||
+      EP.isStoryboardPath(targets[0]) ||
+      EP.isStoryboardChild(targets[0])
+    ) {
+      // TODO Maybe avoid root elements?
+      return []
+    }
+
+    // Is the selected element an only child?
+    const siblings = MetadataUtils.getSiblings(canvasState.startingMetadata, target)
+    if (siblings.length > 1) {
+      return []
+    }
+
+    // Is the selected element a flow layout element?
+    const targetMetadata = MetadataUtils.findElementByElementPath(
+      canvasState.startingMetadata,
+      target,
+    )
+    const isStaticLayout = !(
+      MetadataUtils.isPositionAbsolute(targetMetadata) ||
+      MetadataUtils.isPositionRelative(targetMetadata) ||
+      MetadataUtils.isParentYogaLayoutedContainerForElementAndElementParticipatesInLayout(
+        targetMetadata,
+      )
+    )
+
+    if (!isStaticLayout) {
+      return []
+    }
+
+    // TODO Should we also check isParentZeroSized || isParentContiguous?
+
+    // Time to offer up available strategies for the parent
+    const parentPath = EP.parentPath(target)
+    const ancestorTargetPaths = targetPaths([parentPath])
+    const adjustedCanvasState: InteractionCanvasState = {
+      ...canvasState,
+      interactionTarget: ancestorTargetPaths,
+    }
+
+    // Avoid a cyclic dependency by explicitly passing the other metastrategies when creating the next layer's meta strategy
+    const nextAncestorResult = ancestorMetaStrategy(allOtherStrategies, level + 1)(
+      adjustedCanvasState,
+      interactionSession,
+      customStrategyState,
+    )
+
+    if (nextAncestorResult.length > 0) {
+      // A length of > 0 means that we should be bubbling up to the next ancestor
+      return nextAncestorResult
+    } else {
+      // Otherwise we should stop at this ancestor and return the strategies for this ancestor
+      return allOtherStrategies.flatMap((metaStrategy) =>
+        metaStrategy(adjustedCanvasState, interactionSession, customStrategyState).map((s) => ({
+          ...s,
+          id: `${s.id}_ANCESTOR_${level}`,
+          name: applyLevelSuffix(s.name, level),
+          fitness: s.fitness > 0 ? s.fitness + 10 : s.fitness, // Ancestor strategies should always take priority
+          apply: (strategyLifecycle) => {
+            const result = s.apply(strategyLifecycle)
+            if (result.status === 'success' && result.commands.length > 0) {
+              return {
+                ...result,
+                commands: [
+                  ...result.commands,
+                  highlightElementsCommand([parentPath]),
+                  setCursorCommand('mid-interaction', CSSCursor.MovingMagic),
+                ],
+              }
+            } else {
+              return result
+            }
+          },
+        })),
+      )
+    }
+  }
+}
+
+function applyLevelSuffix(name: string, level: number): string {
+  // FIXME What should we use for the label here?
+  const newSuffix = `(Up ${level})`
```

Comment:
This makes me think the name should be some kind of type which describes the level it is at, so we can just add one to the level rather than repeatedly strip and append the suffix.

Reply:
Yeah I was wondering this. The name for all strategies could be created by their factory functions. Alternatively though it maybe would be enough for the name to be `${name} (ancestor)`. I think this is something to tweak and test out after merging this PR.